### ISSUE TITLE
Build failed on RN version 0.69.1

### ISSUE THREAD

**Issue by Willham12** [th3rdwave/react-native-safe-area-context#294]
```
node_modules/react-native-codegen/lib/parsers/typescript/index.js:191
        throw parsingErrors[0];
        ^

UnusedModuleTypeScriptInterfaceParserError: Module NativeSafeAreaContext.d: Unused NativeModule spec. Please load the NativeModule by calling TurboModuleRegistry.get<Spec>('<moduleName>').

**Comment by antoncojocaru** [comment]
+1

**Comment by Skullcan** [comment]
I'm having the same problem.
Anyone found a solution for this?

**Comment by antoncojocaru** [comment]
Did you create a new project with version 0.69.1 or update an existing project with an older react-native version?

**Comment by Willham12** [comment]
It's an updated version. Looks good with a diff.

**Comment by Skullcan** [comment]
> Did you create a new project with version 0.69.1 or update an existing project with an older react-native version?

I'm using a fresh npx react-native init project.

Steps to reproduce:
- create new project using npx react-native init
- change to the new architecture, make sure its building correctly
- add react-native-safe-area-context and try to rebuild with npx react-native android.

```
E:\projetos\reactnice\node_modules\react-native-codegen\lib\parsers\typescript\index.js:191
        throw parsingErrors[0];
        ^

UnusedModuleTypeScriptInterfaceParserError: Module NativeSafeAreaContext.d: Unused NativeModule spec. Please load the NativeModule by calling TurboModuleRegistry.get<Spec>('<moduleName>').
```

I suspect this is related to the [react-native-code-gen](https://reactnative.dev/docs/new-architecture-app-intro#install-react-native-codegen)
Gonna try to build it w/o the lib in the project after lunch and report back.

**Comment by antoncojocaru** [comment]
If you start a new project with 0.69 you don't need to do thats [changes](https://reactnative.dev/docs/new-architecture-app-intro) in your project. 
To enable new architecture on android you can, set the newArchEnabled property to true by either:
- Changing the corresponding line inside the android/gradle.properties file.
- Set an environment variable ORG_GRADLE_PROJECT_newArchEnabled=true
- Invoke Gradle with -PnewArchEnabled=true

And nothing more.


If you, like us, make changes from [Prerequisites for Applications](https://reactnative.dev/docs/new-architecture-app-intro) from official documentation for new 0.69 project, or  you upgrade to 0.69 from previous versions (with new acrhitecture enabled) you need to remove from <your-project-dir>/android/app/build.gradle this lines:

`apply plugin: "com.facebook.react"`

`react {
    reactRoot = rootProject.file("../node_modules/react-native/")
    codegenDir = rootProject.file("../node_modules/react-native-codegen/")
}`

And this: 
`implementation project(":ReactAndroid") `

replace with:

`implementation "com.facebook.react:react-native:+"`

**Comment by antoncojocaru** [comment]
I think you should try  to install react-native-codegen (0.69.1 version)
And after, run` ./gradlew clean` (inside android project folder)

**Comment by Skullcan** [comment]
> I think you should try to install react-native-codegen (0.69.1 version) And after, run` ./gradlew clean` (inside android project folder)

Hi, I tried using the react-native-codegen 0.69.1 version, same error.
Tried some fixes I saw in the react-native-gesture-handler lib in this [PR](https://github.com/software-mansion/react-native-gesture-handler/pull/2126) too. Sadly no success building it.

I also tried to build it under a WSL2 instance I have running here. Got pretty much the same error in all attempts.
> Android NDK: ERROR:/home/skull/projetos/reactnice/node_modules/react-native-safe-area-context/android/src/main/jni/Android.mk:fb: LOCAL_SRC_FILES points to a missing file
  Android NDK: Check that /home/skull/projetos/reactnice/android/app/build/react-ndk/exported/arm64-v8a/libfb.so exists  or that its path is correct

I'm at a loss atm. Dunno what I'm doing wrong hahaha

**Comment by SudarshJain** [comment]
does anyone solved it?? am getting the same

**Comment by mechaadi** [comment]
Any updates on this issue? Anyone fixed it?

**Comment by tomekzaw** [comment]
> does anyone solved it?? am getting the same

> Any updates on this issue? Anyone fixed it?

Hey, the solution from https://github.com/th3rdwave/react-native-safe-area-context/issues/294#issuecomment-1185682181 works for me! 🥳 

In my case I needed to remove the following lines in `android/app/build.gradle`:

```diff
-apply plugin: "com.facebook.react"
```

```diff
-react {
-    reactRoot = rootProject.file("../node_modules/react-native/")
-    codegenDir = rootProject.file("../node_modules/react-native-codegen/")
-    jsRootDir = rootProject.file("../")
-    libraryName = "appname"
-    codegenJavaPackageName = "com.appname"
-}
```

---

When it comes to the `libfb.so` error, one solution is to simply build the app again. Usually it works on the second attempt. If you CI fails, you can add a step to run the task `cd android && ./gradlew :ReactAndroid:packageReactNdkDebugLibs` that will compile `libfb.so`. Then the app should build as normal.

**Comment by omars94** [comment]
anyone found a solution for this still getting the same error

> [CXX1405] error when building with ndkBuild using node_modules/react-native-safe-area-context/android/src/main/jni/Android.mk: Build command failed.

**Comment by Willham12** [comment]
Compiled with and released an app with RN 0.70.1.
